File: Classes/AreaHourlyCellV1.m
//
//  AreaHourlyCell.m
//  climbingweather
//
//  Created by Jonathan StJohn on 1/30/11.
//  Copyright 2011 __MyCompanyName__. All rights reserved.
//

#import "AreaHourlyCellV1.h"


@implementation AreaHourlyCellV1

@synthesize dayLabel;
@synthesize timeLabel;
@synthesize tempLabel;
@synthesize skyLabel;
@synthesize precipLabel;
@synthesize windLabel;
@synthesize humLabel;
@synthesize conditionsLabel;
@synthesize iconImage;

- (id)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier {
    
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    if (self) {
		
        dayLabel = [[UILabel alloc] initWithFrame: CGRectZero];
		[[self contentView] addSubview: dayLabel];
		
        timeLabel = [[UILabel alloc] initWithFrame: CGRectZero];
		[[self contentView] addSubview: timeLabel];
		
        tempLabel = [[UILabel alloc] initWithFrame: CGRectZero];
		[[self contentView] addSubview: tempLabel];
		
        skyLabel = [[UILabel alloc] initWithFrame: CGRectZero];
		[[self contentView] addSubview: skyLabel];
		
        precipLabel = [[UILabel alloc] initWithFrame: CGRectZero];
		[[self contentView] addSubview: precipLabel];
		
        windLabel = [[UILabel alloc] initWithFrame: CGRectZero];
		[[self contentView] addSubview: windLabel];
		
        humLabel = [[UILabel alloc] initWithFrame: CGRectZero];
		[[self contentView] addSubview: humLabel];
		
        conditionsLabel = [[UILabel alloc] initWithFrame: CGRectZero];
		[[self contentView] addSubview: conditionsLabel];
		
        iconImage = [[UIImageView alloc] initWithFrame: CGRectZero];
		[[self contentView] addSubview: iconImage];
		[iconImage setContentMode: UIViewContentModeScaleAspectFit];
    }
    return self;
}

- (void) layoutSubviews
{
	[super layoutSubviews];
	
	CGRect bounds = [[self contentView] bounds];
	//float h = bounds.size.height;
	float w = bounds.size.width;
	//float valueWidth = 200.0;
	
	float inset = 5.0;
	float columnSpacing = 5.0;
	
	float big = 16.0;
	float small = 12.0;
	
	float timeRowHeight = 30.0;
	float conditionsRowHeight = 13.0;
	float tempRowHeight = 17.0;
	float skyRowHeight = 13.0;
	
	float secondRowY = inset + 20.0;
	float thirdRowY = secondRowY + 20.0;
	
	float timeX = inset;
	float timeY = timeRowHeight / 2.0 - big / 2.0;
	float timeWidth = 65.0;
	
	float iconX = timeX + timeWidth + columnSpacing;
	float iconWidth = 30.0;
	
	float highX = iconX + iconWidth + columnSpacing;
	float highWidth = 70.0;
	
	float precipX = highX + highWidth + columnSpacing;
	float precipWidth = 50.0;
	
	float windX = precipX + precipWidth + columnSpacing;
	float windWidth = 60.0;
	

	
	// Day
	/*
	CGRect innerFrameDay = CGRectMake(dayX, inset, dayWidth, big + 1.0);
	[dayLabel setFont: [UIFont systemFontOfSize: big]];
	[dayLabel setTextAlignment: UITextAlignmentCenter];
	[dayLabel setFrame: innerFrameDay];
	 */
	
	// Time
	CGRect innerFrameTime = CGRectMake(timeX, timeY, timeWidth, timeRowHeight);
    timeLabel.textAlignment = NSTextAlignmentCenter;
	[timeLabel setFont: [UIFont systemFontOfSize: small]];
	[timeLabel setFrame: innerFrameTime];
	
	// Icon
	CGRect innerFrameIcon = CGRectMake(iconX, inset, iconWidth, iconWidth);
	[iconImage setFrame: innerFrameIcon];
	
	// Temp
	CGRect innerFrameTemp = CGRectMake(highX, inset, highWidth, tempRowHeight);
	[tempLabel setFont: [UIFont systemFontOfSize: big]];
    tempLabel.textAlignment = NSTextAlignmentCenter;
	[tempLabel setFrame: innerFrameTemp];
	
	// Sky
	CGRect innerFrameSky = CGRectMake(highX, secondRowY, highWidth, skyRowHeight);
	[skyLabel setFont: [UIFont systemFontOfSize: small]];
    skyLabel.textAlignment = NSTextAlignmentCenter;
	[skyLabel setFrame: innerFrameSky];
	
	// Precip Day
	CGRect innerFramePrecip = CGRectMake(precipX, timeY, precipWidth, timeRowHeight);
	[precipLabel setFont: [UIFont systemFontOfSize: big]];
    precipLabel.textAlignment = NSTextAlignmentCenter;
	[precipLabel setFrame: innerFramePrecip];
	
	// Wind
	CGRect innerFrameWind = CGRectMake(windX, inset, windWidth, tempRowHeight);
	[windLabel setFont: [UIFont systemFontOfSize: big]];
    windLabel.textAlignment = NSTextAlignmentCenter;
	[windLabel setFrame: innerFrameWind];
	
	// Humidity
	CGRect innerFrameHum = CGRectMake(windX, secondRowY, windWidth, skyRowHeight);
	[humLabel setFont: [UIFont systemFontOfSize: small]];
    humLabel.textAlignment = NSTextAlignmentCenter;
	[humLabel setFrame: innerFrameHum];
	
	// Conditions
	CGRect innerFrameConditions = CGRectMake(timeX + 5.0, thirdRowY, w, conditionsRowHeight);
	[conditionsLabel setFont: [UIFont systemFontOfSize: small]];
    conditionsLabel.textAlignment = NSTextAlignmentLeft;
	[conditionsLabel setFrame: innerFrameConditions];
	
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    
    [super setSelected:selected animated:animated];
    
    // Configure the view for the selected state.
}




@end

File: Cells/AreaHourlyCellV1.swift
//
//  AreaHourlyCell.swift
//  climbingweather
//
//  Created by Jon St. John on 3/3/17.
//
//

import Foundation
import UIKit

class AreaHourlyCellV1: UITableViewCell {
    
    let dayLabel = UILabel(frame: CGRect.zero)
    let timeLabel = UILabel(frame: CGRect.zero)
    let tempLabel = UILabel(frame: CGRect.zero)
    let precipLabel = UILabel(frame: CGRect.zero)
    let skyLabel = UILabel(frame: CGRect.zero)
    let windLabel = UILabel(frame: CGRect.zero)
    let humLabel = UILabel(frame: CGRect.zero)
    let conditionsLabel = UILabel(frame: CGRect.zero)
    let iconImage = UIImageView(frame: CGRect.zero)
    
    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
        
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        
        self.contentView.addSubview(self.dayLabel)
        self.contentView.addSubview(self.timeLabel)
        self.contentView.addSubview(self.tempLabel)
        self.contentView.addSubview(self.precipLabel)
        self.contentView.addSubview(self.windLabel)
        self.contentView.addSubview(self.skyLabel)
        self.contentView.addSubview(self.humLabel)
        self.contentView.addSubview(self.conditionsLabel)
        self.iconImage.contentMode = .scaleAspectFill
        self.contentView.addSubview(self.iconImage)
        
        
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        
        let w = self.contentView.bounds.size.width
        
        let inset = 5.0
        let columnSpacing = 5.0
        
        let timeRowHeight = 30.0
        let conditionsRowHeight = 13.0
        let tempRowHeight = 17.0
        let skyRowHeight = 13.0
        
        let big = 16.0
        let small = 12.0
        
        let secondRowY = inset + 20.0
        let thirdRowY = secondRowY + 20.0
        
        let timeX = inset
        let timeY = timeRowHeight / 2.0 - big / 2.0
        let timeWidth = 65.0

        let iconX = timeX + timeWidth + columnSpacing
        let iconWidth = 30.0
        
        let highX = iconX + iconWidth + columnSpacing
        let highWidth = 70.0
        
        let precipX = highX + highWidth + columnSpacing
        let precipWidth = 50.0
        
        let windX = precipX + precipWidth + columnSpacing
        let windWidth = 60.0
        
        // Time
        timeLabel.font = UIFont.systemFont(ofSize: CGFloat(small))
        timeLabel.textAlignment = .center;
        timeLabel.frame = CGRect(x: timeX, y: timeY, width: timeWidth, height: timeRowHeight)
        
        // Icon
        iconImage.frame = CGRect(x: iconX, y: inset, width: iconWidth, height: iconWidth)
        
        // Temp
        tempLabel.font = UIFont.systemFont(ofSize: CGFloat(big))
        tempLabel.textAlignment = .center
        tempLabel.frame = CGRect(x: highX, y: inset, width: highWidth, height: tempRowHeight)
        
        // Sky
        skyLabel.font = UIFont.systemFont(ofSize: CGFloat(small))
        skyLabel.textAlignment = .center
        skyLabel.frame = CGRect(x: highX, y: secondRowY, width: highWidth, height: skyRowHeight)
        
        // Precip
        precipLabel.font = UIFont.systemFont(ofSize: CGFloat(big))
        precipLabel.textAlignment = .center
        precipLabel.frame = CGRect(x: precipX, y: timeY, width: precipWidth, height: timeRowHeight)
        
        // Wind
        windLabel.font = UIFont.systemFont(ofSize: CGFloat(big))
        windLabel.textAlignment = .center
        windLabel.frame = CGRect(x: windX, y: inset, width: windWidth, height: tempRowHeight)
        
        // Humidity
        humLabel.font = UIFont.systemFont(ofSize: CGFloat(small))
        humLabel.textAlignment = .center
        humLabel.frame = CGRect(x: windX, y: secondRowY, width: windWidth, height: skyRowHeight)
        
        // Conditions
        conditionsLabel.font = UIFont.systemFont(ofSize: CGFloat(small))
        conditionsLabel.textAlignment = .left
        conditionsLabel.frame = CGRect(x: timeX + 5.0, y: thirdRowY, width: Double(w), height: conditionsRowHeight)
        
        
    }
    
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}
